internal/exec/terraform_utils.go

```diff
@@ -25,9 +25,36 @@ func checkTerraformConfig(atmosConfig schema.AtmosConfiguration) error {
 // We delete the file to prevent the Terraform prompt asking to select the default or the
 // previously used workspace. This happens when different backends are used for the same component.
 func cleanTerraformWorkspace(atmosConfig schema.AtmosConfiguration, componentPath string) {
-	filePath := filepath.Join(componentPath, ".terraform", "environment")
-	u.LogDebug(fmt.Sprintf("\nDeleting Terraform environment file:\n'%s'", filePath))
-	_ = os.Remove(filePath)
+	// Get TF_DATA_DIR, default to .terraform if not set
+	tfDataDir := os.Getenv("TF_DATA_DIR")
+	if tfDataDir == "" {
+		tfDataDir = ".terraform"
+	}
+
+	// Convert relative path to absolute
+	if !filepath.IsAbs(tfDataDir) {
+		tfDataDir = filepath.Join(componentPath, tfDataDir)
+	}
+
+	// Ensure the path is cleaned properly
+	tfDataDir = filepath.Clean(tfDataDir)
+
+	// Construct the full file path
+	filePath := filepath.Join(tfDataDir, "environment")
+
+	// Check if the file exists before attempting deletion
+	if _, err := os.Stat(filePath); err == nil {
+		u.LogDebug(fmt.Sprintf("Terraform environment file found. Proceeding with deletion.\nFile: %s", filePath))
```

Comment:
```suggestion
		l.LogDebug("Terraform environment file found. Proceeding with deletion.", "file", filePath)
```

Reply:
this wont work

Reply:
We need to use either `u.LogDebug` or `l.Debug`

Reply:
Ok, must be `l.Debug` then